src/main/java/com/marianbastiurea/lifeofbees/Hive.java

```diff
@@ -296,12 +297,12 @@ public void addNewEggsFrameInHive(int year) {
 
 
         int eggsFrameFull = 0;
-        for (int i = 0; i < eggsFrames.size(); i++) {
-            if (this.eggsFrames.get(i).getNumberOfEggs() == maxEggPerFrame && eggsFrameFull < eggsFrames.size()) {
+        for (int i = 0; i < eggFrames.size(); i++) {
+            if (this.eggFrames.get(i).getNumberOfEggs() == this.eggFrames.get(i).getMaxEggPerFrame() && eggsFrameFull < eggFrames.size()) {
                 eggsFrameFull += 1;
             }
         }
-        if (eggsFrameFull == eggsFrames.size()) {
+        if (eggsFrameFull == eggFrames.size()) {
             int maximumNumberOfFramesToAdd = 6 - eggsFrameFull;
             if (maximumNumberOfFramesToAdd != 0) {
                 this.addEggsFrames(createNewEggsFrame());
```

Comment:
Remove `createNewEggsFrame` and `addEggsFrames` methods and just use `this.eggFrames.add(new EggFrame());`

Reply:
done